internal/rest/getv2clustersnamekubeconfig.go

```diff
@@ -46,8 +53,23 @@ func (s *Server) GetV2ClustersNameKubeconfigs(ctx context.Context, request api.G
 		}, nil
 	}
 
-	clusterKubeconfigUpdated, err := updateKubeconfigWithTokenFunc(clusterKubeconfig, namespace, request.Name, request.Params.Authorization)
+	// Determine TTL for kubeconfig JWT based on configuration
+	//    > 0: request a renewed token with specified TTL
+	//   == 0: skip renewal (pass-through existing token)
+	var kubeconfigTTL *time.Duration
+	// always set pointer (including zero) so tokenRenewal can distinguish between "no config provided" (nil) and 0 meaning skip
+	if s.config != nil {
+		kubeconfigTTL = &s.config.KubeconfigTTL
+	}
+
+	clusterKubeconfigUpdated, err := updateKubeconfigWithTokenFunc(clusterKubeconfig, namespace, request.Name, request.Params.Authorization, s.config.DisableAuth, kubeconfigTTL)
 	if err != nil {
+		if strings.Contains(err.Error(), "token expired") || strings.Contains(err.Error(), "token not renewable") {
```

Comment:
Is this for the JWT token in the original GET request? If so, isn't it duplicate check of line 36?

Reply:
the check here is to handle the token that will be embedded in the kubeconfig for authenticating with the downstream cluster, while the first one validates the authorization header of the request to the cluster-manager service itself. So, they have different authn flow